File: 2.2/C++/Symbols.cpp
#include "Symbols.h"

Symbols::Symbols(const std::string& input) : Line(input) {}

size_t Symbols::GetLength() const {
    return value.length();
}

void Symbols::Replace() {
    size_t pos = 0;
    while ((pos = value.find("#", pos)) != std::string::npos) {
        value.replace(pos, 1, "!!");
        pos += 2;
    }
}

File: 2.2/C#/Symbols.cs
namespace Lab
{
    public class Symbols : Line
    {
        public Symbols(string input) : base(input) { }

        public override int GetLength()
        {
            return value.Length;
        }

        public override void Replace()
        {
            value = value.Replace("#", "!!");
        }

    }
}